extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/graal/SubstituteStringUtil.java

```diff
@@ -3,14 +3,35 @@
 import java.security.SecureRandom;
 
 import com.oracle.svm.core.annotate.Alias;
-import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.InjectAccessors;
 import com.oracle.svm.core.annotate.TargetClass;
 
 @TargetClass(className = "liquibase.util.StringUtil")
 final class SubstituteStringUtil {
 
     @Alias
-    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset)
+    @InjectAccessors(SecureRandomAccessors.class)
     private static SecureRandom rnd;
 
+    public static final class SecureRandomAccessors {
+
+        private static volatile SecureRandom volatileRandom;
+
+        public static SecureRandom get() {
+            SecureRandom localVolatileRandom = volatileRandom;
+            if (localVolatileRandom == null) {
+                synchronized (SubstituteStringUtil.class) {
```

Comment:
```suggestion
                synchronized (SecureRandomAccessors.class) {
```

Reply:
I wasn't exactly sure this class would exist in its current form with GraalVM doing its magic. That's why I did it this way.

I can try the other way though and see if it works.

Reply:
It worked so I pushed an update.